### PR TITLE
Changes to improve README.md in SeeDot and respective modifications.

### PR DESCRIPTION
**Pull request by krantikiran68**

### CODE REVIEW

File: tools/SeeDot/SeeDot-dev.py [hunk 1]
```diff
@@ -58,46 +61,58 @@ def parseArgs(self):
...
+                           (Default: 'fixed')")
         parser.add_argument("-d", "--dataset", choices=Dataset.all,
-                            default=Dataset.default, metavar='', help="Dataset to use")
-        parser.add_argument("-m", "--maximisingMetric", choices=config.MaximisingMetric.all, metavar='',
-                            help="What metric to maximise during exploration",default=config.MaximisingMetric.default)
+                            default=Dataset.default, metavar='', help="Dataset to use\
+                            (Default: 'usps10')")
+        parser.add_argument("-m", "--metric", choices=config.Metric.all, metavar='',
+                            help="Select the metric that will be used to measure the correctness of an inference, to obtain the \
+                            best quantization of variables. (valid only for Classification) \
```
Comment (ShikharJ) [inline tools/SeeDot/SeeDot-dev.py]: I believe we can lowercase `Classification` here.

File: tools/SeeDot/seedot/compiler/input/fastgrnn.sd [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
+
+//FastGRNN SeeDot code for usps10 dataset
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/input/fastgrnn.sd]: Can we have a space after the `//` please? And a full stop at the end of the sentence?

File: tools/SeeDot/seedot/compiler/ir/irBuilder.py [hunk 3]
```diff
@@ -16,8 +16,10 @@
 import seedot.compiler.type as Type
 from seedot.util import *
 
-# IRBuilder class converts the input AST into IR, which is a sequence of function calls.
-# Each node in the input grammar is handled with it's own function in this class.
+'''
+IRBuilder class converts the input AST into IR, which is a sequence of function calls.
+Each node in the input grammar is handled with it's own function in this class.
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/ir/irBuilder.py]: `It's` should be `its` here.

File: tools/SeeDot/seedot/compiler/ir/irUtil.py [hunk 4]
```diff
@@ -7,7 +7,9 @@
 
 from seedot.util import *
 
-# Util functions used by IR.py and IRBuilder.py
+'''
+Util functions used by IR.py and IRBuilder.py
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/ir/irUtil.py]: Missing punctuation at the end.

File: tools/SeeDot/seedot/compiler/ir/ir.py [hunk 5]
```diff
@@ -8,16 +8,18 @@
...
-# subst() is used for substitutions, for example:
-#   In the input code, suppose we have a + b.
-#   In the AST, it will be represented as add(var(a), var(b)).
-#   The compiler substitutes 'a' with tmp1.
-#   The substitution tmp1/'a' (read as tmp1 for 'a', or 'a' should be replaced with tmp1) is applied on the whole program.
-#   All the subst() calls are written in a way that within the AST, all instances of 'a' would be replaced by tmp1.
-#   The updated AST would look like add(var(tmp1), var(b)).
+'''
+Primitives used by the Intermediate Representation (IR).
+For most primitives, there would be in __init__() function and one subst() function.
```
Comment (ShikharJ) [inline tools/SeeDot/seedot/compiler/ir/ir.py]: `in` should be replaced with `an` here.

File: tools/SeeDot/architecture.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+On running this command, 
+
+1. `main` function in `SeeDot-dev.py` initiates the `MainDriver` class in `SeeDot-dev.py`, which in turn initiates the `Main` class in `seedot/main.py`. 
+
+2. In `seedot/main.py`, depending on the encoding, the `funForFloat` or `runForFixed` function is called. (In this case, `runForFloat` will be called). 
+
+3. In `runForFloat`, first the `Converter` class in `seedot/compiler/converter/converter.py` is instantiated with the arguments `(encoding=float, datasetType=Testing, target=x86)`. 
+
+    (The arguments here are for representation purposes and to explain the differences between the multiple Converter instances).
```
Comment (ShikharJ) [inline tools/SeeDot/architecture.md]: Converter should be `Converter` here.

File: tools/SeeDot/architecture.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+4. In the `Converter` class, the SeeDot input files (all the model weights and, the train and test datasets) are processed to a format that is required by the SeeDot compiler. 
+
+5. In `seedot/compiler/converter/coverter.py`, the quantization code is called that produces files used by the SeeDot compiler. 
+    Depending on the encoding, `QuantizerFloat` or `QuantizerFixed` classes are instantiated for `float` or `fixed` encoding respectively. 
+    These classes are defined in `seedot/compiler/converter/quantizer.py`. (In our case `QuantizerFloat`).
+
+6. The `QuantizerFloat` class uses the `ParamsBuilder` class from `seedot/compiler/converter/paramsBuilder.py` to quantize the parameters. 
+
+7. After the `Converter` class finishes execution, the control return to `seedot/main.py:runForFloat`.
```
Comment (ShikharJ) [inline tools/SeeDot/architecture.md]: returns in place of return here.

File: tools/SeeDot/architecture.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+2. In `seedot/main.py`, depending on the encoding, the `funForFloat` or `runForFixed` function is called. (In this case, `runForFloat` will be called). 
+
+3. In `runForFloat`, first the `Converter` class in `seedot/compiler/converter/converter.py` is instantiated with the arguments `(encoding=float, datasetType=Testing, target=x86)`. 
+
+    (The arguments here are for representation purposes and to explain the differences between the multiple Converter instances). 
+
+4. In the `Converter` class, the SeeDot input files (all the model weights and, the train and test datasets) are processed to a format that is required by the SeeDot compiler. 
+
+5. In `seedot/compiler/converter/coverter.py`, the quantization code is called that produces files used by the SeeDot compiler.
```
Comment (ShikharJ) [inline tools/SeeDot/architecture.md]: `...called, which...` here.

File: tools/SeeDot/architecture.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    python SeeDot-dev.py -a fastgrnn -e fixed -d usps10 -n 1 -t x86 -m red_disagree
+```
+
+On running this command, 
+
+1. `main` function in `SeeDot-dev.py` initiates the `MainDriver` class in `SeeDot-dev.py`, which in turn initiates the `Main` class in `seedot/main.py`. 
+2. In `seedot/main.py`, depending on the encoding, the `funForFloat` or `runForFixed` function is called. (In this case, `runForFixed` will be called). 
+3. In `runForFixed`, first the `collectProfileData` function is called. This collects the data that helps determines the most efficient scale and bitwidth of variables.
+
+4. In `collectProfileData` the `Converter` class in `seedot/compiler/converter/converter.py` is instantiated with the arguments `(encoding=float, datasetType=Training, target=x86)`. (The arguments here are for representation purposes and to explain the differences between the multiple Converter instances).
```
Comment (ShikharJ) [inline tools/SeeDot/architecture.md]: Please change Converter to `Converter`.

File: tools/SeeDot/architecture.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+On running this command, 
+
+1. `main` function in `SeeDot-dev.py` initiates the `MainDriver` class in `SeeDot-dev.py`, which in turn initiates the `Main` class in `seedot/main.py`. 
+2. In `seedot/main.py`, depending on the encoding, the `funForFloat` or `runForFixed` function is called. (In this case, `runForFixed` will be called). 
+3. In `runForFixed`, first the `collectProfileData` function is called. This collects the data that helps determines the most efficient scale and bitwidth of variables.
+
+4. In `collectProfileData` the `Converter` class in `seedot/compiler/converter/converter.py` is instantiated with the arguments `(encoding=float, datasetType=Training, target=x86)`. (The arguments here are for representation purposes and to explain the differences between the multiple Converter instances). 
+5. In `Converter`, the quantization code is called that produces files used by the SeeDot compiler. Depending on the encoding, `QuantizerFloat` or `QuantizerFixed` classes are instantiated for `float` and `fixed` encoding respectively. These classes are defined in `seedot/compiler/converter/quantizer.py`. (In our case `QuantizerFoat`).
```
Comment (ShikharJ) [inline tools/SeeDot/architecture.md]: `...called, which...` here.

**Review by ShikharJ — COMMENTED** [review]
Minor comments. Otherwise, this looks good to go.

File: tools/SeeDot/SeeDot-dev.py (hunk 1 above)
Comment (krantikiran68) [reply tools/SeeDot/SeeDot-dev.py]: Done

**Review by ShikharJ — APPROVED** [review]

**Pull request merged**